tensorboard/webapp/feature_flag/views/feature_flag_modal_trigger_container.ts

```diff
@@ -0,0 +1,55 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component} from '@angular/core';
+import {MatDialog, MatDialogRef} from '@angular/material/dialog';
+import {Store} from '@ngrx/store';
+import {State} from '../../app_state';
+import {featureFlagOverrideChanged} from '../actions/feature_flag_actions';
+import {
+  getFeatureFlags,
+  getShowFlagsEnabled,
+} from '../store/feature_flag_selectors';
+import {FeatureFlagPageContainer} from './feature_flag_page_container';
+
+@Component({
+  selector: 'feature-flag-modal-trigger',
+  template: ``,
+  styles: [],
+})
+export class FeatureFlagModalTriggerContainer {
+  readonly showFeatureFlags$ = this.store.select(getShowFlagsEnabled);
+  readonly featureFlags$ = this.store.select(getFeatureFlags);
+  private featureFlagsDialog?: MatDialogRef<FeatureFlagPageContainer>;
+
+  constructor(private readonly store: Store<State>, private dialog: MatDialog) {
+    this.showFeatureFlags$.subscribe((showFeatureFlags: boolean) => {
+      if (showFeatureFlags) {
+        this.featureFlagsDialog = this.dialog.open(FeatureFlagPageContainer);
+        this.featureFlagsDialog.afterClosed().subscribe(() => {
+          // By disabling the flag when the dialog is closed to prevent it from appearing again after the page is refreshed.
+          this.store.dispatch(
+            featureFlagOverrideChanged({
+              flags: {enableShowFlags: false},
+            })
+          );
+        });
+        return;
+      }
+      if (this.featureFlagsDialog) {
+        this.featureFlagsDialog.close();
```

Comment:
This doesn't seem right. The only times I can imagine this code path being invoked are:

* User sets showFlags to false in the dialog.
* User presses "Reset All" in the dialog.

In neither case would I expect the dialog to be immediately closed.

Reply:
I think it depends on how you think about `showFlags`. If you really think it is a feature flag then it makes sense that it should close, but I can definitely see why you would view that as confusing so I'll go ahead and remove this.